types/virtual.go

```diff
@@ -0,0 +1,7 @@
+package types
+
+type Lesser interface {
```

Comment:
For bikeshedding `Comparable`

Reply:
`Comparable` goes against the Go conventions of being an "-er" and also breaks the convention of utilizing the name of the method if it makes sense. `Reader`, isn't `Funneler`, or `Gatherer` because the one method is called `Read`. :)

However, this is related to the above discussion of being able to support `slices.SortFunc` so, I'll defer to the outcome of that.

Reply:
Yes and no, there's the `comparable` interface in Go now.  Though I think your suggestion is more correct.

https://go.dev/ref/spec#Type_constraints

All that said, it probably doesn't matter much, as I think we can have the `Lesser` interface be internal or private so it's not something the general public has to know about.

Reply:
`comparable` is only for builtin types that have _all of the comparison_ operators defined for them. It's not a 1 method interface. :)

Reply:
As we discussed in DM, since this interface embeds `Value`, I think calling it `ComparableValue` is a fair name. With both `Less()` and `LessEqual()` on the interface, I think a user can implement any comparison they want.

Also, I hate when -`er` is appended to a non-verb or non-verb phrase. What in the world is a `Stringer` (aside from a character from HBO's The Wire) anyway?